File: src/main/java/org/jukeboxmc/form/FormListener.java
package org.jukeboxmc.form;

import lombok.Getter;
import org.jetbrains.annotations.NotNull;

import java.util.function.Consumer;

/**
 * @author GoMint
 * @version 1.0
 */
@Getter
public class FormListener<R> {

    private Consumer<R> responseConsumer = r -> { };
    private Consumer<Void> closeConsumer = aVoid -> { };

    public @NotNull FormListener<R> onResponse(Consumer<R> consumer ) {
        this.responseConsumer = consumer;
        return this;
    }

    public @NotNull FormListener<R> onClose(Consumer<Void> consumer ) {
        this.closeConsumer = consumer;
        return this;
    }

}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/form/FormListener.kt
package org.jukeboxmc.api.form

import java.util.function.Consumer

/**
 * @author Kaooot
 * @version 1.0
 */
class FormListener<R> {

    private var responseConsumer: Consumer<Any> = Consumer { }
    private var closeConsumer: Consumer<Void?> = Consumer { }

    fun onResponse(consumer: Consumer<Any>): FormListener<R> {
        this.responseConsumer = consumer
        return this
    }

    fun onClose(consumer: Consumer<Void?>): FormListener<R> {
        this.closeConsumer = consumer
        return this
    }

    fun getResponseConsumer() = this.responseConsumer

    fun getCloseConsumer() = this.closeConsumer
}